### ISSUE TITLE
Jetson AGX Orin: Nemotron 3.5 Lightning with reproducible W4A16/Q4 quantization

### ISSUE THREAD

**Issue by OriNachum** [agentculture/lobes-cli#211]
## Announcement

Add a **Jetson AGX Orin 64GB** deployment path for **NVIDIA Nemotron 3.5 Lightning 30B-A3B**, but do **not** make an unverifiable third-party quant the reference artifact.

The canonical source of truth should be NVIDIA's full-precision post-trained checkpoint:

`nvidia/NVIDIA-Nemotron-3.5-Lightning-30B-A3B-BF16`

From that source, produce a locally reproducible Orin-oriented quantization artifact and record the complete provenance needed to reproduce it.

This is the Orin counterpart to #187, which is intentionally Thor-specific.

## Why Orin needs its own path

Jetson AGX Orin is Ampere (SM87). It does not have native FP4 tensor-core execution.

NVIDIA's official Nemotron 3.5 Lightning NVFP4 checkpoint is nevertheless documented to run on Ampere through **W4A16** kernels. That makes W4A16 the performance-oriented vLLM path for Orin, while native NVFP4 remains a Blackwell concern.

For llama.cpp, a self-produced **Q4_K_M GGUF** is the comparison/fallback path.

Do not conflate:

- **stored checkpoint format** (`NVFP4`, GGUF Q4, etc.);
- **compute path** (`W4A16`, native FP4, BF16);
- **runtime** (vLLM / llama.cpp / TensorRT-LLM).

The advert must report these separately where possible.

## Artifact trust / provenance requirement

Existing community W4A16/GGUF conversions may be useful as references, but they must not become the default unless their source revision and quantization recipe are independently reproducible.

For the reference Orin artifact, record at minimum:

- upstream model: `nvidia/NVIDIA-Nemotron-3.5-Lightning-30B-A3B-BF16`;
- exact Hugging Face revision / commit SHA;
- source file hashes or snapshot identity;
- quantizer/tool and exact version/commit;
- complete quantization command/config;
- calibration dataset + revision + sample count/sequence length, when calibration is used;
- modules intentionally excluded or kept at higher precision;
- output artifact hashes;
- target runtime/container revision;
- hardware used for quantization and validation.

The build should be reproducible through `agentculture/unsloth-cli` rather than a manual one-off conversion.

### Path A — W4A16 for vLLM on AGX Orin

Primary performance target.

Use NVIDIA Model Optimizer behind `unsloth-cli` to generate a ModelOpt-compatible **W4A16** artifact from the official BF16 checkpoint.

Current NVIDIA ModelOpt exposes weight-only 4-bit schemes intended for Ampere-or-later deployment, including:

- `w4a16_nvfp4` / `W4A16_NVFP4_CFG` — FP4 weights, BF16 activations;
- `int4_blockwise_weight_only` — INT4 weights, BF16 activations;
- `int4_awq` / W4A16 AWQ where supported by the architecture/runtime.

For this model, do not assume the generic recipe is automatically optimal. Validate Nemotron-H / Mamba-2 + MoE support and compare against NVIDIA's own published mixed-precision Nemotron recipe/checkpoint behavior.

Important: this path should use **ModelOpt**, even though the orchestration command lives in `unsloth-cli`. Upstream Unsloth's GGUF export is not itself a vLLM/ModelOpt W4A16 exporter.

Candidate interface:

```bash
sloth quantize \
  --model nvidia/NVIDIA-Nemotron-3.5-Lightning-30B-A3B-BF16 \
  --target jetson-agx-orin \
  --format modelopt \
  --quant w4a16 \
  --output artifacts/nemotron-3.5-lightning-orin-w4a16
```

The command should resolve this to a pinned ModelOpt recipe and print/store the exact underlying command.

### Path B — Q4_K_M GGUF for llama.cpp

Produce our own GGUF from the same official BF16 source as an independently reproducible comparison/fallback artifact.

Upstream Unsloth currently exposes GGUF export with `q4_k_m`; llama.cpp is therefore the natural backend here.

Candidate interface:

```bash
sloth quantize \
  --model nvidia/NVIDIA-Nemotron-3.5-Lightning-30B-A3B-BF16 \
  --target jetson-agx-orin \
  --format gguf \
  --quant q4_k_m \
  --output artifacts/nemotron-3.5-lightning-orin-q4-k-m
```

Also allow Q5_K_M/Q8_0 only for quality/performance comparison; Q4_K_M should be the first Orin benchmark.

## `unsloth-cli` work required

`agentculture/unsloth-cli` currently focuses on LoRA/QLoRA train/eval/export. Add a standalone **quantization** workflow that does not require fine-tuning first.

Suggested contract:

```text
sloth quantize
  --model <hf-id-or-local-path>
  --revision <sha>
  --target jetson-agx-orin
  --format modelopt|gguf
  --quant w4a16|q4_k_m|...
  --output <path>
  --dry-run
  --json
```

Requirements:

1. Pin and persist the upstream HF revision before downloading/quantizing.
2. Refuse to label an artifact reproducible if the source revision is floating.
3. ModelOpt backend for Orin W4A16; Unsloth/llama.cpp backend for GGUF.
4. Emit a machine-readable manifest next to every artifact.
5. Manifest must include source revision, tool/container versions, underlying command, recipe/calibration metadata and SHA-256 hashes.
6. `--dry-run` must show the exact container and quantization command without spending GPU time.
7. Keep quantization separate from fine-tuning; `sloth export` should not silently become a generic PTQ command.
8. Do not upload artifacts by default.
9. If pushing to Hugging Face is later enabled, require explicit repo id and preserve provenance manifest.
10. Make hardware-specific presets declarative so Thor/Orin/Spark can diverge without model-name conditionals.

## Runtime validation on AGX Orin 64GB

Benchmark both self-produced artifacts and, as a control, NVIDIA's official NVFP4 checkpoint running through its documented Ampere W4A16 path.

### vLLM / W4A16

Start from NVIDIA's Ampere recipe, adapted for Orin memory/concurrency rather than copying A100 batch settings literally:

- ModelOpt quantized checkpoint;
- W4A16 execution;
- FlashInfer Mamba backend where supported;
- prefix caching;
- realistic `--max-model-len` (start at 32K/64K, then increase);
- reasoning parser `nemotron_v3`;
- tool parser compatible with the model;
- measure plain decoding before speculative decoding.

### llama.cpp / Q4_K_M

Validate:

- model loads completely on AGX Orin 64GB;
- CUDA offload is complete/expected;
- chat template/reasoning behavior is correct;
- tool calling is not advertised unless actually validated;
- MTP/DFlash only if supported by the exact llama.cpp build and checkpoint combination.

## Benchmark matrix

On the same AGX Orin 64GB, compare:

1. NVIDIA official `...-NVFP4` served on the documented Ampere/W4A16 path;
2. our self-quantized W4A16 artifact from NVIDIA BF16;
3. our self-quantized Q4_K_M GGUF from NVIDIA BF16.

Record:

- model/source revision;
- artifact size;
- resident memory;
- TTFT;
- output tok/s at batch/concurrency 1;
- throughput at small practical concurrency;
- prompt processing speed;
- quality sanity/eval delta vs BF16 or official NVIDIA artifact;
- tool-call validity;
- context sizes that actually fit;
- startup time;
- runtime/container versions;
- thermal/power mode used on Orin.

Do not compare headline tok/s numbers produced under different speculative-decoding or concurrency settings without labeling the difference.

## Acceptance criteria

- [ ] Orin 64GB gets a first-class Nemotron 3.5 Lightning profile separate from Thor #187.
- [ ] Reference source is NVIDIA's official BF16 post-trained checkpoint pinned to an exact revision.
- [ ] `unsloth-cli` can reproducibly build the selected Orin W4A16 artifact through a ModelOpt backend.
- [ ] `unsloth-cli` can reproducibly build a Q4_K_M GGUF from the same BF16 source.
- [ ] Both artifacts have complete machine-readable provenance manifests and hashes.
- [ ] No unverifiable third-party quant is required for the reference deployment.
- [ ] W4A16 and Q4_K_M are benchmarked on the same AGX Orin 64GB.
- [ ] NVIDIA's official NVFP4 checkpoint on its Ampere/W4A16 execution path is included as a control.
- [ ] lobes advert reports source/quantization/runtime truthfully and does not imply native FP4 execution on Orin.
- [ ] A chat probe and tool-call probe pass before the model is advertised ready.
- [ ] The selected default is based on measured Orin performance + quality, not only checkpoint size.

## Related

- #187 — Thor Nemotron 3.5 Lightning worker deployment
- `agentculture/unsloth-cli` — reproducible quantization/build tooling
- NVIDIA BF16 source: `nvidia/NVIDIA-Nemotron-3.5-Lightning-30B-A3B-BF16`
- NVIDIA optimized checkpoint/control: `nvidia/NVIDIA-Nemotron-3.5-Lightning-30B-A3B-NVFP4`

## Non-goals

- Pretending Orin has native FP4 tensor-core execution.
- Making a community conversion authoritative merely because it loads.
- Treating GGUF Q4 and vLLM W4A16 as equivalent formats.
- Replacing the Thor-specific work in #187.
- Fine-tuning Nemotron as part of this ticket; this is post-training quantization and serving.
